src/Middleware/MiddlewareChainerMixin.php

```diff
@@ -43,9 +43,11 @@
      *
      * @param MiddlewareChainer $next The middleware to append
      *
-     * @return MiddlewareChainer A new middleware queue with $next appended
+     * @return MiddlewareChainerMixin A new middleware queue with $next appended
      */
-    public function withMiddleware(MiddlewareChainer $next) :MiddlewareChainer
+    public function withMiddleware(
+        MiddlewareChainer $next
+    ): MiddlewareChainerMixin
```

Comment:
This change is not correct. This trait is a helper for implementing the MiddlewareChainer interface, which returns a MiddlewareChainer, not a MiddlewareChainerMixin.

Reply:
Phan seems to think it returns a Mixin. Should the warning be suppressed?

Reply:
That's only because traits aren't allowed to declare that they implement an interface in PHP (even though in cases like this one it fully implements it), and PHP requires interfaces to be explicitly implemented with the `implements` keyword.

Is this the only thing preventing one of the currently ignored classes of warnings from being added to phan's config/Travis?

Reply:
I reverted my changes to the file

Reply:
Until it's the last error and holding up the check being automated I don't think there any reason to suppress it, that would just make it more likely to be missed if something were to change that makes it fixable (such as a hypothetical future version of PHP that allowed traits to implement interfaces, or a bug fix to phan that does a better job dealing with this type of situation.)